genesyscloud/flow_loglevel/resource_genesyscloud_flow_loglevel_utils.go

```diff
@@ -0,0 +1,83 @@
+package flow_loglevel
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v125/platformclientv2"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+)
+
+func generateFlowLogLevelResource(
+	flowId string,
+	flowLoglevel string,
+	resourceId string,
+) string {
+	return fmt.Sprintf(`resource "genesyscloud_flow_loglevel" "%s" {
+	  flow_id					= "%s"
+	  flow_log_level 			= "%s"
+	}`,
+		resourceId,
+		flowId,
+		flowLoglevel)
+}
+
+// getFlowLogLevelSettingsRequestFromResourceData maps data from schema ResourceData object to a platformclientv2.Flowloglevelrequest
+func getFlowLogLevelSettingsRequestFromResourceData(d *schema.ResourceData) platformclientv2.Flowloglevelrequest {
+	return platformclientv2.Flowloglevelrequest{
+		LogLevelCharacteristics: getFlowLogLevelFromResourceData(d),
+	}
+}
+
+// getFlowLogLevelFromResourceData maps data from schema ResourceData object to a platformclientv2.Flowloglevel
+func getFlowLogLevelFromResourceData(d *schema.ResourceData) *platformclientv2.Flowloglevel {
```

Comment:
I think you can change the logic of this function just to make it shorter. First you can create the `platformclientv2.Flowloglevel` struct with the `Level`. After that do your check to see if you need to add the `Characteristics`, you don't need an else that way and can just have one return statement. P.S. There's a function built into our go sdk that can return a pointer to a string, you can use it for required attributes, e.g. 
```
logLevel := platformclientv2.Flowloglevel{
	Level:   platformclientv2.String(d.Get("flow_log_level").(string)),
}
```